scripts/add_changelog.py

```diff
@@ -0,0 +1,132 @@
+    " urgency=(?P<urgency>\w+).*"
+)
+CHANGELOG_FILE = "debian/changelog"
+GIT_GBP_CONF = ".git/gbp.conf"
+PACKAGE_GBP_CONF = "debian/gbp.conf"
+PACKAGE_GBP_CUSTOMIZATION = "debian/gbp_format_changelog"
+
+DEFAULT_GBP_CONF = os.path.join(os.path.dirname(__file__), "gbp.conf")
+DEFAULT_GBP_CUSTOMIZATION = os.path.join(
+    os.path.dirname(__file__), "gbp_format_changelog"
+)
+
+
+def get_parser():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("message", help="Changelog message to add")
+    parser.add_argument("version", help="Packaging version for changelog")
+    parser.add_argument(
+        "include_bugs",
+        help='Set "true" to include bugs in changelog',
+        choices=["true", "false"],
+    )
+    return parser
+
+
+def _get_gbp_env():
+    """Return a dict of gbp-related environment variables."""
+    if not any(
+        [
+            os.path.exists(PACKAGE_GBP_CONF),
+            os.path.exists(GIT_GBP_CONF),
+        ]
+    ):
+        print(
+            f"NOTICE: no gbp.conf found in debian/ or .git/."
+            f" Using: {DEFAULT_GBP_CONF}"
+        )
+        return {"GBP_CONF_FILES": DEFAULT_GBP_CONF}
+    return None
+
+
+def add_changelog(msg: str, version: str, include_bugs: str = "false"):
+    with open(CHANGELOG_FILE) as stream:
+        full_changelog = stream.read().splitlines()
+    unreleased_snapshot_messages = []  # Upstream snapshot commit messages
+    changelog_lines = []
+    pkg_info = re.match(PKG_RELEASE_RE, full_changelog[0]).groupdict()
+    unreleased_commitish = None
+    if pkg_info["dist"] == "UNRELEASED":
+        # Reconstruct top changelog entry
+        _, _, pkg_commitish = pkg_info["version"].partition("g")
+        if len(pkg_commitish) == 8:  # Then we have the commitish
+            orig_commitish = pkg_commitish
+        found_snapshot = False
+        changelog_count = 0
+        for line in full_changelog:
+            if re.match(
+                rf"{pkg_info['pkg_name']} .*urgency={pkg_info['urgency']}", line
+            ):
+                changelog_count += 1
+            if NEW_UPSTREAM_MSG in line and changelog_count == 1:
+                found_snapshot = True
+                continue
+            if found_snapshot and changelog_count == 1:
+                if line and not line.startswith(" -- "):
+                    if NEW_UPSTREAM_MSG not in line:
+                        unreleased_snapshot_messages.append(
+                            re.sub(r"^ +\+ ?", "+", line)
+                        )
+                else:
+                    changelog_lines.append(line)
+            else:
+                changelog_lines.append(line)
+        if changelog_lines != full_changelog:
+            with open(CHANGELOG_FILE, "w") as stream:
+                stream.write("\n".join(changelog_lines) + "\n")
+    # Add current msg comment
+    gbp_cmd = ["gbp", "dch", "--ignore-branch", f"--new-version={version}"]
```

Comment:
Maybe we should point out somewhere in the documentation that `gpb` must be installed. Or add something like
`gpb --version || error "gpb is required. Please run 'sudo apt install git-buildpackage'"` at the top of  `new-upstram-snapshot`.

Otherwise, `new-upstram-snapshot` is going to fail at the middle of the execution and left the user with an inconsistent repo state, most probably.

Reply:
+1 I've added the following to main
```python
    try:
        check_output(["gbp", "--help"])
    except:
        print(
            f"ERROR: {os.path.basename(__file__)} requires gbp. Run:"
            " sudo apt install git-buildpackage"
        )
        sys.exit(1)
```